diff --git a/libsensors/MagSensor.cpp b/libsensors/MagSensor.cpp
index 48933ff..758ef54 100755
--- a/libsensors/MagSensor.cpp
+++ b/libsensors/MagSensor.cpp
@@ -28,8 +28,8 @@
 #include <cutils/properties.h>
 #include "MagSensor.h"
 
-#define MAG_CTRL_NAME    "FreescaleMagnetometer"
-#define MAG_DATA_NAME    "eCompass" 
+#define MAG_DATA_NAME    "FreescaleMagnetometer"
+#define MAG_CTRL_NAME    "eCompass"
 #define MAG_SYSFS_PATH   "/sys/class/input"
 #define MAG_SYSFS_DELAY  "poll"
 #define MAG_SYSFS_ENABLE "enable"
@@ -324,7 +324,7 @@ int MagSensor::sensor_get_class_path(char *class_path)
 		    continue;
 		}
 		buf[res - 1] = '\0';
-		if (strcmp(buf, MAG_CTRL_NAME) == 0) {
+		if (strcmp(buf, MAG_DATA_NAME) == 0) {
 		    found = 1;
 		    close(fd);
 		    break;
